src/opt/fuzz_test.rs

```diff
@@ -198,7 +198,7 @@ fn prove_equiv(expr: Expr, optimized: Expr, tcx: &TyCtx) -> TestCaseResult {
         optimized.clone(),
     );
     let ctx = z3::Context::new(&z3::Config::new());
-    let smt_ctx = SmtCtx::new(&ctx, tcx);
+    let smt_ctx = SmtCtx::new(&ctx, tcx, false, false);
```

Comment:
These additional parameters are rather mysterious here. I suggest just adding a new type like `SmtCtxOptions`to capture those options.